test/Concurrency/Runtime/async_task_async_let_child_cancel.swift

```diff
@@ -0,0 +1,59 @@
+// RUN: %target-run-simple-swift(-Xfrontend -enable-experimental-concurrency %import-libdispatch -parse-as-library) | %FileCheck %s --dump-input=always
+
+// REQUIRES: executable_test
+// REQUIRES: concurrency
+
+func printWaitPrint(_ int: Int) async -> Int {
+  print("start, cancelled:\(Task.isCancelled), id:\(int)")
+  while !Task.isCancelled {
+    await Task.sleep(100_000)
+  }
```

Comment:
this sleeps/spins until cancelled, and notice that if cancellation wouldnt go through child records this would spin forever -- (and I confirmed it did spin forever before this fix)